tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,370 @@
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/configmap"
+	"github.com/openshift-kni/eco-goinfra/pkg/mco"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/cmd"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/define"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/frrconfig"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/netenv"
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/netinittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/security/internal/tsparams"
+	ocpoperatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ignition "github.com/coreos/ignition/v2/config/v3_4/types"
+	"gopkg.in/k8snetworkplumbingwg/multus-cni.v4/pkg/types"
+	apimachinerytype "k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("nftables", Ordered, Label(tsparams.LabelNftablesTestCases), ContinueOnFailure, func() {
+
+	var (
+		hubIPv4ExternalAddresses = []string{"172.16.0.10", "172.16.0.11"}
+		hubIPv4Network           = "172.16.0.0/24"
+		masterPodIPv4Address     = "172.16.0.1"
+		portNum8888              = 8888
+		cnfWorkerNodeList        []*nodes.Builder
+		masterNodeList           []*nodes.Builder
+		ipv4NodeAddrList         []string
+		ip4Worker0NodeAddr       []string
+		ipv4SecurityIPList       []string
+		testPodWorker0           *pod.Builder
+		testPodList              []*pod.Builder
+		routeMap                 map[string]string
+		mcNftablesName           = "98-nftables-cnf-worker"
+		interfaceNameNet1        = "net1"
+		interfaceNameBrEx        = "br-ex"
+		err                      error
+	)
+	BeforeAll(func() {
+		By("List CNF worker nodes in cluster")
+		cnfWorkerNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+		Expect(len(cnfWorkerNodeList)).To(BeNumerically(">", 1),
+			"Failed to detect at least two worker nodes")
+
+		By("Selecting worker node for Security tests")
+		ipv4NodeAddrList, err = nodes.ListExternalIPv4Networks(
+			APIClient, metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to collect external nodes ip addresses")
+
+		ip4Worker0NodeAddr = []string{ipv4NodeAddrList[0]}
+
+		By("Listing master nodes")
+		masterNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+		Expect(len(cnfWorkerNodeList)).To(BeNumerically(">", 1),
+			"Failed to detect at least two worker nodes")
```

Comment:
Ugh?? Why have you changed this? It was fine before your latest change. This is wrong as is.

Reply:
I thought in your previous comment you were asking me to add a check that would pass on a single node.  I am now only testing one node but as I said earlier the test cases are built for multiple nodes in mind. So to be sure which tests work on a single node cluster we will need to check.